plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/ExceptionMappingInputStream.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spooling.filesystem;
+
+import java.io.FileNotFoundException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+public class ExceptionMappingInputStream
+        extends FilterInputStream
+{
+    public ExceptionMappingInputStream(InputStream delegate)
+    {
+        super(delegate);
+    }
+
+    @Override
+    public int read()
+            throws IOException
+    {
+        try {
+            return super.read();
+        }
+        catch (FileNotFoundException e) {
+            throw new RuntimeException("Segment not found or expired", e);
```

Comment:
These should rethrow `IOException`, as it's strange to throw an unchecked exception here